### PR TITLE
Event creation

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-719

### Change description ###
Moved event creation from FileErrorHandler to FileContentProcessor. This makes code cleaner and allows to better understand the workflow.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by lgonczar** [comment]
I'd like to propose an alternative solution: renaming `FileErrorHandler` to `FileRejector`/`FileRejectionService`/`EnvelopeReje...` - this would make it clear what it does. Creating an event would then be a natural part of rejection and doesn't need to be extracted.

**Comment by apereverzin** [comment]
@lgonczar The point of this PR is to make clear where failure event is created.Before it was created in FileContentProcessor and in FileErrorHandler, now only in FileContentHandler, this makes code more clear. I agree with renaming to FileRejector, and also suggest to rename _handleInvalidFileError_ method to _handleInvalidFile_.

**Comment by apereverzin** [comment]
@lgonczar The point of this PR is to make clear where failure event is created.Before it was created in FileContentProcessor and in FileErrorHandler, now only in FileContentHandler, this makes code more clear. I agree with renaming to FileRejector, and also suggest to rename _handleInvalidFileError_ method to _handleInvalidFile_.

**Comment by lgonczar** [comment]
> @lgonczar The point of this PR is to make clear where failure event is created.Before it was created in FileContentProcessor and in FileErrorHandler, now only in FileContentHandler, this makes code more clear. I agree with renaming to FileRejector, and also suggest to rename _handleInvalidFileError_ method to _handleInvalidFile_.

Fair enough

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileContentProcessorTest.java [hunk 1]
```diff
@@ -35,11 +35,12 @@
     private static final String FILE_NAME = "file1.zip";
     private static final String CONTAINER_NAME = "container";
     private static final String LEASE_ID = "leaseID";
-    private static final String DCN = "dcn";
     private static final String POBOX = "pobox";
     private static final String BULKSCAN = "bulkscan";
     private static final String CASE_NUMBER = "case_number";
     private static final String CASE_REFERENCE = "case_reference";
+    private static final Long EVENT_ID = 1L;
+    public static final String MSG = "msg";
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileContentProcessorTest.java:43]: Can we keep this change outside this PR? I don't agree with creating static variables without any particular meaning rather than being used multiple times as any value. We can move that discussion into another PR, not to block this one.
